tests/validations/test_metrics.py

```diff
@@ -664,23 +687,42 @@ def test_cumulative_metrics() -> None:  # noqa: D
                     type_params=PydanticMetricTypeParams(
                         measure=PydanticMetricInputMeasure(name=measure_name),
                         cumulative_type_params=PydanticCumulativeTypeParams(
-                            window=PydanticMetricTimeWindow.parse(
-                                window="3 martian_weeks", custom_granularity_names=("martian_week",), strict=False
-                            ),
+                            window=PydanticMetricTimeWindow.parse(window="3 martian_weeks"),
                         ),
                     ),
                 ),
+                metric_with_guaranteed_meta(
+                    name="custom_grain_to_date",
+                    type=MetricType.CUMULATIVE,
+                    type_params=PydanticMetricTypeParams(
+                        measure=PydanticMetricInputMeasure(name=measure_name),
+                        cumulative_type_params=PydanticCumulativeTypeParams(
+                            grain_to_date="3 martian_weeks",
+                        ),
+                    ),
+                ),
+                metric_with_guaranteed_meta(
+                    name="custom_window_old",
+                    type=MetricType.CUMULATIVE,
+                    type_params=PydanticMetricTypeParams(
+                        measure=PydanticMetricInputMeasure(name=measure_name),
+                        window=PydanticMetricTimeWindow.parse(window="5 martian_week"),
+                    ),
+                ),
             ],
             project_configuration=EXAMPLE_PROJECT_CONFIGURATION,
         )
     )
 
     build_issues = validation_results.all_issues
-    assert len(build_issues) == 3
+    assert len(build_issues) == 7
     expected_substrings = [
         "Invalid time granularity",
         "Both window and grain_to_date set for cumulative metric. Please set one or the other.",
         "Got differing values for `window`",
+        "Invalid time granularity 'martian_week' in window: '3 martian_weeks'",
+        "Invalid time granularity 'martian_week' in window: '3 martian_week'",
+        "Invalid time granularity 'martian_week' in window: '5 martian_week'",
```

Comment:
Do we have tests for successful custom grain parsing now that these fails?

Reply:
Yes on line 339! That one is successful.